components/brave_new_tab_ui/api/topSites.ts

```diff
@@ -3,13 +3,46 @@
 // License, v. 2.0. If a copy of the MPL was not distributed with this file,
 // you can obtain one at http://mozilla.org/MPL/2.0/.
 
-/**
- * Obtains the top sites
- */
-export function getTopSites (): Promise<chrome.topSites.MostVisitedURL[]> {
-  return new Promise(resolve => {
-    chrome.topSites.get((topSites: chrome.topSites.MostVisitedURL[]) => {
-      resolve(topSites || [])
-    })
-  })
+let areCustomLinksEnabled: boolean = true
+let areTopSitesVisible: boolean = true
+
+export function updateMostVisitedInfo () {
+  chrome.send('updateMostVisitedInfo')
+}
+
+export function addMostVistedInfoChangedListener (listener: any): void {
```

Comment:
Please can this define the data that is provided in the listener, instead of `any`? Makes it easier to diagnose and follow the code (not to mention throw errors at compile time if we get it wrong).

e.g.

```ts
type ChangedEventData = {
   isVisible: boolean
   customLinksEnabled: boolean
   tiles: TopSite[]
}
type MostVisitedInfoChangedHandler = (data: ChangedEventData) => any)

export function addMostVistedInfoChangedListener (listener: MostVisitedInfoChangedHandler): void
```

Reply:
FIXED